### PR TITLE
Path as array

### LINKED ISSUE #44: Path should be an array
Path arg should be an array, since prop name could contain `.`

### PR DESCRIPTION
**Pull request by DarrenPaulWright**
Resolves #44 

I started a lib directory to move specific functions to so we can test and benchmark them individually. I intend to move more code there over time.

I also switched the benchmark library to karma-benchmark. matcha is not being maintained and doesn't work properly on recent versions of node.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,11 +10,11 @@
 		"url": "sindresorhus.com"
 	},
 	"engines": {
-		"node": ">=6"
+		"node": ">=10"
```
Comment (sindresorhus) [inline package.json]: I did this as a separate commit to make it clear to us that the next version will require a major bump.

https://github.com/sindresorhus/on-change/commit/b5e2678c48f82eb3ba863e3964d25d261c7fe27b

File: lib/path.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			return path;
+		}
+
+		if (key && key.toString !== undefined) {
+			if (path !== '') {
+				path += PATH_SEPARATOR;
+			}
+
+			if (isSymbol(key)) {
+				return path + 'Symbol(' + key.description + ')';
```
Comment (sindresorhus) [inline lib/path.js]:
```suggestion
				return path + `Symbol(${key.description})`;
```

File: lib/path.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			return '';
+		}
+
+		return path.slice(0, index);
+	},
+	walk: (path, callback) => {
+		if (isArray(path)) {
+			path.forEach(callback);
+		} else if (path !== '') {
+			let pos = 0;
```
Comment (sindresorhus) [inline lib/path.js]:
```suggestion
			let position = 0;
```

More readable to not use abbreviations.

**Comment by sindresorhus** [comment]
Nice work! 👍

File: lib/path.js (hunk 2 above)
Comment (DarrenPaulWright) [reply lib/path.js]: The benchmarks for this function drop over 10% when I make this change,  so I would rather keep it if that's alright with you.

Comment (sindresorhus) [reply lib/path.js]: Up to you. I would argue that if the benchmarks are affected by this, the benchmarks are faulty and focusing too much on microbenchmarking and not on real-world scenarios. Also make sure you're testing in the latest Node.js version.

**Comment by sindresorhus** [comment]
1.8.0 should have been a major release (2.0.0) as requiring Node.js 10 is a breaking change.

**Comment by DarrenPaulWright** [comment]
Ok. Should I publish a 1.8.1 that reverts the changes, then 2.0.0?

**Comment by sindresorhus** [comment]
Yes

**Comment by DarrenPaulWright** [comment]
1.8.1 is live. Before I push 2.0.0, should I deal with the 2 TODOs that refer to the next major release?

**Comment by DarrenPaulWright** [comment]
Or could you? I'm not really sure about the TODO in index.t.ds

**Comment by sindresorhus** [comment]
Done: https://github.com/sindresorhus/on-change/commit/5f4251b0b2e5b33c50df11e5d74092818e4cb3e2

**Pull request merged**